### PR TITLE
Auto Cloak Units

### PR DESCRIPTION
**Pull request by Ezahiel**
Auto Cloak Units widget loaded by default, for new in game settings, with 1 new file(unit_auto_cloak.lua) and updated 4(gui_options.lua, en.lua, armamex.lua, armckfus.lua). Please do not merge until @Ruwetuin   or myself fix issue with config save/load. Making PR on @Ruwetuin  request.

### CODE REVIEW

**Comment by Ezahiel** [comment]
Those are re-worked Cloak and new Options
screen00092

**Comment by Sovgut** [comment]
@Ezahiel awesome!

File: luaui/Widgets/gui_options.lua [hunk 1]
```diff
@@ -3685,6 +3685,260 @@ function init()
...
+			description = texts.option.auto_cloak_armckfus_descr,
+			onload = function(i)
+				loadWidgetData("Auto Cloak Units", "auto_cloak_armckfus", { "armckfus" })
+			end,
+			onchange = function(i, value)
+				saveOptionValue("Auto Cloak Units", "init_cloak", "set_ArmCkfus", { "armckfus" }, value)
+			end,
+		},
+
+		{ id = "auto_cloak_armspy", group = "game", category = types.basic,
```
Comment (sprunk) [inline luaui/Widgets/gui_options.lua:3909]: most of those are the same, maybe autogenerate them

File: luaui/Widgets/unit_auto_cloak.lua [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+end
+
+function widget:GetConfigData()
+    spEcho("[Auto Cloak Units] widget:GetConfigData (default: "..tostring(enabled).."".."******".."armjamt: "..tostring(cloaking_Unit_Default[UnitDefNames["armjamt"].id])..")") -- DEBUG
+    return {
+	    enabled = enabled,
+        armjamt = cloaking_Unit_Default[UnitDefNames["armjamt"].id],
+        armdecom = cloaking_Unit_Default[UnitDefNames["armdecom"].id],
+        cordecom = cloaking_Unit_Default[UnitDefNames["cordecom"].id],
+        armferret = cloaking_Unit_Default[UnitDefNames["armferret"].id],
```
Comment (sprunk) [inline luaui/Widgets/unit_auto_cloak.lua:78]: use a loop

File: luaui/Widgets/unit_auto_cloak.lua [hunk 3]
```diff
@@ -0,0 +1,232 @@
+
+local info = {
+    name      = "Auto Cloak Units",
+    desc      = "Auto cloaks Units with Cloak",
+    author    = "wilkubyk",
+    date      = "2022.12.29",
+    license   = "GNU GPL, v2 or later",
+    layer     = math.huge, --someone check if could be changed
```
Comment (WatchTheFort) [inline luaui/Widgets/unit_auto_cloak.lua:8]: Layer is limited to ±1,000,000

**Review by Ezahiel — COMMENTED** [review]
well done , love it @Ruwetuin

**Pull request merged**